packages/@aws-cdk/aws-ecr-assets/test/test.image-asset.ts

```diff
@@ -212,19 +213,38 @@ export = {
     const app = new App();
     const stack = new Stack(app, 'stack');
 
-    const image = new DockerImageAsset(stack, 'MyAsset', {
-      directory: path.join(__dirname, 'dockerignore-image')
-    });
+    const directory = fs.mkdtempSync(path.join(os.tmpdir(), 'dockerignore-image'));
+    createFsStructureFromTree(directory, `
```

Comment:
Oh that is clever and pretty sexy!

What I was thinking when I made my previous remark though is testing the exclude pattern evaluator separately from the `listFilesRecursively` operation. So I was thinking something like:

```ts
const shouldInclude = parseExcludePattern(['foobar.txt']); 
// or whatever API we choose, doesn't have to directly return a function, maybe it is a class but
// it is something that should give me a yes/no answer

test.ok(shouldInclude('Dockerfile'));
test.ok(shouldInclude('subdirectory/baz.txt'));
test.ok(!shouldInclude('foobar.txt'));
test.ok(!shouldInclude('subdirectory/foobar.txt'));
```

I can then test various edge case behaviors of the exclude pattern evaluator in-memory (speaking of `fs`-calls efficiency!) and then I only have to have 1 or 2 tests, maybe they're already the integ tests, to make sure that `listFilesRecursively` uses the exclude pattern evaluator in the right way.

Reply:
This will only be possible if we resolve the main-thread discussion about needing the `fs` to evaluate the exclusion list:

```ts
const shouldInclude = parseExcludePattern(['dir', '!dir/file']); 

// naively, this would be false
// but we would need to create dir if `dir/file` exists
test.ok(shouldInclude('dir'));

test.ok(shouldInclude('dir/file'));
test.ok(!shouldInclude('dir/other_file'));
```